src/cpp/shared_core/include/shared_core/system/User.hpp

```diff
@@ -101,6 +101,13 @@ class User
     */
    User(const User& in_other);
 
+   /**
+    * @brief Move constructor.
+    *
+    * @param in_other   The user to move into this User.
+    */
+   User(User&& in_other) noexcept;
```

Comment:
I wonder if we could just use the default compiler-implemented move constructor here?